### DISCUSSION TITLE
The best way to send an email notification when message was delivered?

### DISCUSSION THREAD

**Discussion by kenzotakahashi** [supabase/supabase#8025] [Questions] [answered]
I'm building an app that allows people to send messages to each other. Just a basic app.
I want to notify the recipient via email.

How would I do that?

I use mandrill for email provider and `@mailchimp/mailchimp_transactional` for client library.
I confirmed that I can send emails from my Nuxt app, but obviously that wouldn't be secure because I'm exposing mandrill key.
I would like to know how to send emails from backend.

I learned about edge functions, but I couldn't figure out how to do it.

And if edge function is the way to go, can I use it on production now?

Thanks in advance!

**Comment by dshukertjr** [comment, marked as answer]
Hi @kenzotakahashi,

If you are using Vercel or Netlify to deploy your Nuxt app, you can use their functions to host your backend that will contain the logic to send our emails. You can certain use our edge functions as well. 

In order to send realtime emails whenever there was a change in the database, you need webhooks. For that, you can use our [Database Webhooks](https://supabase.com/blog/2021/07/30/supabase-functions-updates#database-webhooks-alpha). With database webhooks, you can call your backend whenever certain change happen on your database like when a user has registered, or when a user has received a comment.

**Reply by kenzotakahashi** [reply]
@dshukertjr 
Thanks for the answer!
I was using Vercel but didn't know about its serverless functions. They are exactly what I needed!
I also didn't know that you live in Tokyo. That's awesome. Hi neighbor ✋

**Reply by dshukertjr** [reply]
@kenzotakahashi 
Hi to you too neighbor 😉 Glad I was able to help!
